src/backend/web/routes/feeds.js

```diff
@@ -51,4 +52,19 @@ feeds.get('/:id', async (req, res) => {
   }
 });
 
+feeds.post('/', protect, async (req, res) => {
+  const { newFeed, email } = req.body;
+  try {
+    /* check if email has a key in redis
+  if not create key
+  if user has a key in redis, check if feed already exists, (not sure how fast this is? or if its necessary), if not store feed in feeds.
```

Comment:
>not sure how fast this is?

From what I understand, Redis is super fast (plus can fetch that info in constant time), so we probably don't have to worry about speed in this use case.

>or if its necessary

Also from what I understand, it is impossible to add a duplicate item to Redis set. I'm not sure yet what the end result of such an attempt is (I'm guessing an error being thrown, which you've set up handling for, here...). So, I infer that it's not necessary to implement a uniqueness check, but going to defer to @c3ho on this one.

Reply:
Yeah, Redis won't error, but will just overwrite the value of the key-value 
`redis-cli set hi bye` will create a key called `hi` with the value `bye`
`redis-cli get hi` will return `bye`
`redis-cli set hi hi` will overwrite the current value of `bye` with `hi`
`redis-cli get hi` will now return `hi`

Reply:
so if i'm not mistaken the key will be linked to the user? so, what if I post a feed, and say Rafi does to a blog, would that duplicate the blog feed? (I only ask because I do believe some school staff have a seneca email and a seneca work email)

Reply:
t:users will contain
`t:user:grommers00@myseneca.ca:feeds`
`t:user:silvyre@myseneca.ca:feeds`

`t:user:grommers00@myseenca.ca:feeds` can contain
`t:feed:a3e61 (a feed for OSD600)` <-- not too sure about the exact wording or way, but it will be implemented something like this
`t:feed:a3eb1 (a feed for OSD700)`

The key should be linked to a specific user


>so, what if I post a feed, and say Rafi does to a blog, would that duplicate the blog feed? (I only ask because I do believe some school staff have a seneca email and a seneca work email)

Not sure what you mean by this, but we could have a duplication if a user signs up with their seneca email and also a seneca work email as we'll have two users now
`t:user:grommers00@myseneca.ca:feeds`
`t:user:grommers00@work.myseneca.ca:feeds`

Reply:
`t:users` will contain `grommers00@myseneca.ca` and to get the user info, you'd create a full key from that "id": `t:user:grommers00@myseneca.ca`.  We probably don't need the `:feeds` suffix, since that's the only thing we'll store for a user (i.e., their list of feeds).

Then, `t:user:grommers00@myseneca.ca` will be a Set that contains just the id for each of the feeds that this user owns.  To get those feeds, you do `t:feed:{id}`

Reply:
@humphd Makes sense!